(dp1
S'output'
p2
S"<type 'exceptions.NameError'> global name 'Device' is not defined"
p3
sS'layer'
p4
S'/home/abdulhalim/Desktop/web2py/applications/motor/controllers/default.py'
p5
sS'code'
p6
S'<code object <module> at 0x7f2f61ba34b0, file "/home/abdulhalim/Desktop/web2py/applications/motor/controllers/default.py", line 2>'
p7
sS'snapshot'
p8
(dp9
sS'traceback'
p10
S'Traceback (most recent call last):\n  File "/home/abdulhalim/Desktop/web2py/gluon/restricted.py", line 219, in restricted\n    exec(ccode, environment)\n  File "/home/abdulhalim/Desktop/web2py/applications/motor/controllers/default.py", line 126, in <module>\n  File "/home/abdulhalim/Desktop/web2py/gluon/globals.py", line 409, in <lambda>\n    self._caller = lambda f: f()\n  File "/home/abdulhalim/Desktop/web2py/applications/motor/controllers/default.py", line 68, in get_instruction\n    db.Status.update_or_insert((db.Status.device_ref_id==Device.id) & (db.Device.device_id==device_id), last_ping=ping, server_time=ping)\nNameError: global name \'Device\' is not defined\n'
p11
s.